lsoa/views.py

```diff
@@ -614,3 +617,162 @@ class ListTag(BaseTagManagement, ListView):
+        if not name_tuples:
+            return Student.objects.none()
+
+        query = reduce(operator.or_, (Q(first_name=f, last_name=l) for f, l in name_tuples))
+
+        return Student.objects.filter(query)
+
+    def get_context_data(self):
+        data = super().get_context_data()
+        objects = data['object_list']
+        grouped_data = dict()
+        for object in objects:
+            name = ' '.join([object.first_name, object.last_name])
+            if name in grouped_data:
+                grouped_data[name].append(object)
+            else:
+                grouped_data[name] = [object]
+        data['object_list'] = grouped_data
+        return data
+
+
+class StudentReportAjax(View):
+
+    def validate_floating(self, student_id, action):
+        try:
+            student = Student.objects.get(pk=student_id)
+        except Student.DoesNotExist:
+            return False, None
+
+        if action not in ['active', 'inactive']:
+            return False, None
+
+        return True, student
+
+    def validate_doubled(self, student_id, action, course_ids):
+        try:
+            student = Student.objects.get(pk=student_id)
+        except Student.DoesNotExist:
+            return False, None
+
+        if action not in ['split']:
+            return False, None
+
+        count = Course.objects.filter(pk__in=course_ids, students__in=[student_id]).count()
+        if count != len(course_ids):
+            return False, None
+
+        return True, student
+
+    def validate_homonym(self, student_ids, name, action):
+        students = Student.objects.filter(pk__in=student_ids)
+        is_valid = True
+
+        if not (student_ids and name and action):
+            is_valid = False
+
+        elif any(slugify(student.name) != name for student in students):
+            is_valid = False
+
+        elif action not in ['merge']:
+            is_valid = False
+
+        return is_valid, list(students)
+
+    def post(self, request):
+        self.request = request
+        report = request.POST.get('report', '')
+
+        if report == 'floating':
+            student_id = request.POST.get('student_id', '')
+            action = request.POST.get('action', '')
+            is_valid, student = self.validate_floating(student_id, action)
+            if is_valid:
+                student.status = action
+                student.save()
+                return JsonResponse({
+                    'success': True,
+                    'status': student.status
+                })
+
+        elif report == 'doubled':
+            student_id = request.POST.get('student_id', '')
+            action = request.POST.get('action', '')
+            course_ids = request.POST.getlist('course_ids[]', [])
+
+            is_valid, student = self.validate_doubled(student_id, action, course_ids)
+
+            if is_valid and student:
+                all_courses = Course.objects.filter(students__in=[student])
```

Comment:
You fan do `filter(students__pk=student.pk)`, without doing this workaround with a list.